### PR TITLE
Update to Webpack 4 and turn on OptimizeCssAssetsPlugin

### PR DESCRIPTION
**Pull request by julianguyen**
# Description 

Update to Webpack 4 and turn on OptimizeCssAssetsPlugin

extract-text-webpack-plugin is deprecated for Webpack 4. Tried the recommended mini-css-extract-plugin, but there was no hot reloading. Tried extract-css-chunks-webpack-plugin and it worked :)

- [x] Viewing site works
- [x] Running tests works
- [x] Viewing Storybook works
- [x] Hot reloading of Storybook works
- [x] Hot reloading of site works

Do code splitting in another PR 🐼 

## Corresponding Issue

https://github.com/ifmeorg/ifme/issues/970

# Test Coverage

🚫  N/A

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -45,39 +44,36 @@
...
-    "@storybook/addon-actions": "^3.4.2",
-    "@storybook/addon-backgrounds": "^3.4.2",
-    "@storybook/addon-info": "^3.4.2",
-    "@storybook/addons": "^3.4.2",
-    "@storybook/react": "^3.4.2",
+    "@babel/plugin-proposal-class-properties": "^7.1.0",
+    "@babel/preset-env": "^7.1.0",
+    "@babel/preset-flow": "^7.0.0",
+    "@babel/preset-react": "^7.0.0",
+    "@storybook/addon-actions": "^4.0.0-alpha.21",
```
Comment (julianguyen) [inline client/package.json]: Only version that supports Webpack 4

File: app/assets/stylesheets/dashboard/dashboard_nav.scss [hunk 2]
```diff
@@ -7,7 +7,7 @@
   }
 
   &Links {
-    @include setFontSize($size-16);
+    @include setFontSize($size-18);
```
Comment (julianguyen) [inline app/assets/stylesheets/dashboard/dashboard_nav.scss]: LOL this was too small for my liking

File: app/views/shared/_dashboard_nav_actions.html.erb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 <div class="dashboardNavActions">
-  <%= link_to(t('shared.header.signout'), destroy_user_session_path, method: :delete, class: 'buttonGhostM') %>
+  <%= link_to(t('shared.header.signout'), destroy_user_session_path, method: :delete, class: 'buttonGhostS') %>
```
Comment (julianguyen) [inline app/views/shared/_dashboard_nav_actions.html.erb]: This button looked massive compared to the other items in the menu

**Comment by julianguyen** [comment]
Ahh this is weird I'm now getting this locally:
```
00:00:59 web.1    | Error during failsafe response: Webpacker can't find webpack_bundle.css in /Users/julia/ifme/public/webpack/manifest.json. Possible causes:
00:00:59 web.1    | 1. You want to set webpacker.yml value of compile to true for your environment
00:00:59 web.1    |    unless you are using the `webpack -w` or the webpack-dev-server.
00:00:59 web.1    | 2. webpack has not yet re-run to reflect updates.
00:00:59 web.1    | 3. You have misconfigured Webpacker's config/webpacker.yml file.
00:00:59 web.1    | 4. Your webpack configuration is not creating a manifest.
00:00:59 web.1    | Your manifest contains:
00:00:59 web.1    | {
00:00:59 web.1    |   "webpack_bundle.js": "/webpack/webpack_bundle-b576f364233fc2b4ff67.js"
```

:/ even when I go back to commits where the tests when green

File: client/app/startup/registration.jsx [hunk 4]
```diff
@@ -3,7 +3,7 @@
  * Add components only if needed in current application, because
  * this bundle will be loaded in production.
  */
-
+import '@babel/polyfill';
```
Comment (baohouse) [inline client/app/startup/registration.jsx]: You can move this into the "entry" webpack parameter. e.g.
https://github.com/baohouse/labs.baohouse.net/blob/master/webpack.config.js

Comment (julianguyen) [reply client/app/startup/registration.jsx]: Ooh you're right, thanks for the callout

**Pull request merged**